cmd/up.go

```diff
@@ -118,11 +123,29 @@ List of components that kubeapps up installs:
 			c.SkipGc = false
 		}
 
-		if err = c.Run(objs, wd); err != nil {
-			return fmt.Errorf("can't install kubeapps components: %v", err)
+		// mongodb secret
+		// FIXME (tuna): if the mongodb secret exists then do nothing,
+		// otherwise, add it (with new generated rand pw) to the objs list then do full update.
+		if prevsecret, exist, err := mongoSecretExists(c, MongoDB_Secret, Kubeapps_NS); err != nil {
+			return err
+		} else if !exist {
+			pwFields := []string{"mongodb-password", "mongodb-root-password"}
+			pw := make(map[string]string)
+			for _, p := range pwFields {
+				s, err := generateEncodedRandomPassword(12)
+				if err != nil {
+					return fmt.Errorf("error reading random data for secret %s: %v", MongoDB_Secret, err)
+				}
+				pw[p] = s
+			}
+			secret := buildMongoDBSecret(pw, MongoDB_Secret, Kubeapps_NS)
+			objs = append(objs, secret)
+		} else if exist {
+			// add prevsecret to the list so it won't be GC-ed
+			objs = append(objs, prevsecret)
```

Comment:
why adding the element to the list? what `GC-ed` means?

Reply:
It's garbage collector. `kubeapps up` relies on kubecfg's physiology - it maintains list of objects to be deployed/updated/removed together. Because we create the secret object separately, we should add it into the list so it will also be managed by kubecfg. Correct me if I understand it wrong @anguslees